neurox/data/extraction/transformers_extractor.py

```diff
@@ -272,11 +272,19 @@ def extract_representations(
     print("Reading input corpus")
 
     def corpus_generator(input_corpus_path):
-        with open(input_corpus_path, "r") as fp:
-            for line in fp:
-                yield line.strip()
+        if not isinstance(input_corpus_path, list) and os.path.exists(input_corpus_path):
+            with open(input_corpus_path, "r") as fp:
+                for line in fp:
+                    yield line.strip()
+                return
+        else:
+            if isinstance(input_corpus_path,str):
```

Comment:
when is this condition used? do we expect the user to pass a single "sentence" as the input corpus?